homeassistant/components/sunsynkweb/sensor.py

```diff
@@ -0,0 +1,183 @@
+"""Sensors for the sunsynk web api."""
+
+import decimal
+import logging
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import PERCENTAGE, UnitOfEnergy, UnitOfPower
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from . import DOMAIN
+from .coordinator import PlantUpdateCoordinator
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class SunSynkApiSensor(CoordinatorEntity, SensorEntity):
+    """Parent class for sunsynk api exposing sensors.
+
+    The sensors expose the sum of the power across all inverters
+    for plants that have more than one inverter.
+    State of charge is normally shared between inverters, so this will take
+    the maximum of the state of charge.
+    """
+
```

Comment:
```suggestion
    _attr_has_entity_name = True
```
required for new integrations